File: SeaweedChat.Domain/Aggregates/MessageAggregate/Message.cs
namespace SeaweedChat.Domain.Aggregates;

public class Message : Entity
{
    public DateTime CreatedAt { get; set; }
    public DateTime? EditAt { get; set; } = null;
    public string? Text { get; set; }
    public User Owner { get; set; } = null!;
    public Chat Chat { get; set; } = null!;
}

File: SeaweedChat.Web/src/types/api/message.ts
import type { Chat } from "./chat";

export interface MessageCreateParams {
  chat: Chat;
  text: string;
}
export interface MessagesGetParams {
  chat: Chat;
  offset?: number;
}
export interface MessageDeleteParams {
  id: string;
  chatId: string;
}
export interface MessageEditParams {
  id: string;
  chatId: string;
  text: string;
}
export interface Message {
  id: string;
  text: string;
  ownerId: string;
  ownerUsername: string;
  createdAt: Date;
  editAt: Date;
}
